packages/io-auth-n-identity-commons/src/utils/storage-blob.ts

```diff
@@ -0,0 +1,140 @@
+import { BlobServiceClient } from "@azure/storage-blob";
+import { pipe } from "fp-ts/lib/function";
+import * as E from "fp-ts/Either";
+import * as O from "fp-ts/Option";
+import * as TE from "fp-ts/TaskEither";
+
+/**
+ * Converts a ReadableStream into a string.
+ *
+ * @param readable - The ReadableStream to convert.
+ * @returns A TaskEither that resolves to the string content of the stream,
+ *          or an Error on failure.
+ */
+export const streamToText = (
+  readable: NodeJS.ReadableStream,
+): TE.TaskEither<Error, string> =>
+  pipe(
+    TE.tryCatch(async () => {
+      readable.setEncoding("utf8");
+
+      // eslint-disable-next-line functional/no-let
+      let result = "";
+      for await (const chunk of readable) {
+        result += chunk.toString();
+      }
+      return result;
+    }, E.toError),
+  );
+
+/**
+ * Checks if a blob exists in the specified Azure Blob Storage container.
+ *
+ * @param blobServiceClient - The Azure BlobServiceClient instance used to interact with Blob Storage.
+ * @param containerName - The name of the container where the blob is stored.
+ * @param blobName - The name of the blob to check for existence.
+ * @returns A TaskEither that resolves to true if the blob exists, false otherwise,
+ *          or an Error on failure.
+ */
+export const blobExists = (
+  blobServiceClient: BlobServiceClient,
+  containerName: string,
+  blobName: string,
+): TE.TaskEither<Error, boolean> =>
+  pipe(
+    TE.tryCatch(
+      () =>
+        blobServiceClient
+          .getContainerClient(containerName)
+          .getBlobClient(blobName)
+          .exists(),
+      E.toError,
+    ),
+  );
+
+/**
+ * Downloads a blob from the specified Azure Blob Storage container and returns its readable stream.
+ *
+ * @param blobServiceClient - The Azure BlobServiceClient instance used to interact with Blob Storage.
+ * @param containerName - The name of the container where the blob is stored.
+ * @param blobName - The name of the blob to be downloaded.
+ * @returns A TaskEither that resolves to the blob's readable stream on success,
+ *          or a NotFoundError if the blob does not exist, or an InternalError on other failures.
+ */
+const downloadBlob =
+  (blobServiceClient: BlobServiceClient, containerName: string) =>
+  (blobName: string): TE.TaskEither<Error, NodeJS.ReadableStream> =>
+    pipe(
+      TE.tryCatch(
+        () =>
+          blobServiceClient
+            .getContainerClient(containerName)
+            .getBlobClient(blobName)
+            .download(),
+        E.toError,
+      ),
+      TE.chainW((response) =>
+        TE.fromEither(
+          pipe(
+            response.readableStreamBody,
+            O.fromNullable,
+            E.fromOption(() => Error("Blob stream is null or undefined")),
+          ),
+        ),
+      ),
```

Comment:
nit: you can simplify in this way!
```suggestion
      TE.chainW(({ readableStreamBody }) =>
        TE.fromNullable(Error("Blob stream is null or undefined"))(
          readableStreamBody,
        ),
      ),
```